### PR TITLE
Implicitly-typed lambda parameters have RefKind.None

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/64985

> The parameters of an anonymous function in the form of a _lambda_expression_ can be explicitly or implicitly typed. In an explicitly typed parameter list, the type of each parameter is explicitly stated. In an implicitly typed parameter list, the types of the parameters are inferred from the context in which the anonymous function occurs—specifically, when the anonymous function is converted to a compatible delegate type or expression tree type, that type provides the parameter types (§10.7).

> The parameter list of an anonymous function in the form of an _anonymous_method_expression_ is optional. If given, the parameters shall be explicitly typed. If not, the anonymous function is convertible to a delegate with any parameter list not containing out parameters.

### CODE REVIEW

**Comment by 333fred** [comment]
I don't agree with the original bug here. Is the syntax illegal? Yes. However, the user's experience is likely to be better if we respect the `ref` that was written in source, as cascaded errors (such as lambda applicability or ref mismatches) are more likely to be lined up with the `ref` that the user actually specified. I don't think we should change our behavior here.

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/DelegateTests.cs [hunk 1]
```diff
@@ -712,9 +712,6 @@ class C
     D d = async delegate { };
 }";
             CreateCompilation(source).VerifyDiagnostics(
-                // (4,17): error CS1988: Async methods cannot have ref, in or out parameters
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/DelegateTests.cs:715]: This error should not be removed.

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/DelegateTests.cs [hunk 2]
```diff
@@ -712,9 +712,6 @@ class C
...
-                //     D d = async delegate { };
-                Diagnostic(ErrorCode.ERR_BadAsyncArgType, "delegate"),
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/DelegateTests.cs:717]: From offline discussion, @333fred pointed out we should still have an error in this scenario. I agree.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs [hunk 3]
```diff
@@ -15635,7 +15635,7 @@ void M()
             Assert.Null(model.GetSymbolInfo(invocations[1]).Symbol);
         }
 
-        [Fact]
+        [Fact, WorkItem(64985, "https://github.com/dotnet/roslyn/issues/64985")]
```
[the remark this answers is not in the document]
Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs]: Removed

**Review by 333fred — APPROVED** [review]

**Pull request merged**